crowbar_framework/app/models/service_object.rb

```diff
@@ -1852,4 +1852,41 @@ def node_changed_roles?(node, old_role, new_role)
     roles_in_new = new_role.elements.keys.select { |r| new_role.elements[r].include?(node) }.sort
     roles_in_old != roles_in_new
   end
+
+  # return true if no other attribute than the ignored has changed. And that the node has no other
+  # role than the roles in only_for_roles
+  def relevant_attributes_changed_if_roles?(node, old_role, new_role, ignore_attr, only_for_roles)
+    # get roles for this node
+    roles_in_new = new_role.elements.keys.select { |r| new_role.elements[r].include?(node) }.sort
+
+    # return true if node has other roles that ones in the list (only_for_roles)
+    return true unless Set.new(roles_in_new).subset?(Set.new(only_for_roles))
+
+    # prepare a clone of default attributes of old and new roles
+    old_selected_attributes = old_role.default_attributes[@bc_name].clone
+    new_selected_attributes = new_role.default_attributes[@bc_name].clone
```

Comment:
Asking @Itxaka: `clone` vs `deep_dup`: what's the right one?

Reply:
With clone we'll have a shallow copy where if there are a reference, it copies the reference to the same object. On the other hand, if we use deep_dup a new object will be created for this reference.
If will be references, and we don't want to change the information of any of them I guess must use deep_dup.
I am going to do tests